src/components/chatbot/ChatBox.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react'
+import ChatBoxForm from './ChatBoxForm';
+import ChatMessage from './ChatMessage';
+import { useChat } from '@/hooks/useChat';
+import { useSendMessage } from '@/hooks/useBot';
+import { ByUser, Message } from '@/types/messages/message';
+import { getMessagesFromChat } from '@/lib/chat';
+
+type P = {
+  // ID of the chat to retrieve history of messages from. Not the TG Chat ID, but the actual chat Id stored in database
+  chatId: number;
+}
+
+export default function ChatBox({chatId}: P) {
+  const {data: chat, isPending} = useChat(chatId);
+  const sendMessageMutation = useSendMessage(chatId)
+
+  const [messages, setMessages] = useState<Message[]>([])
+  
+  useEffect(() => {
+    if (chat) {
+      setMessages(getMessagesFromChat(chat));
+    }
+
+  }, [chat])
+  
+  useEffect(() => {
+    try {
+      // TODO: Replace URL with constant string
+      const socket = new WebSocket('ws://localhost:8080/api/ws');
+  
+      socket.onopen = () => {
+        console.log('WebSocket connection established');
+      };
+  
+      socket.onmessage = (event) => {
+        setMessages((prevMessages) => [...prevMessages, event.data])
+        console.log('Message from server: ', event.data);
+      };
+  
+      socket.onclose = () => {
+        console.log('WebSocket connection closed');
+      };
+  
+      socket.onerror = (ev) => {
+        console.error('error event: ', ev);
+      };
+      return () => {
+        socket.close();
+      };
+    } catch (error) {
+      console.error(error)
+    }
+
+  }, [])
+
+  const onSubmit = async (values: any) => {
```

Comment:
_**any**_

You can use `z.infer<typeof formSchema>`. To avoid cyclic imports, we can have a separate folder for all the schema definition and validation logic

Reply:
I think cyclic imports are not an issue in React, so there shouldn't be a need to have a separate folder to house all that